ipaserver/plugins/trust.py

```diff
@@ -446,8 +446,13 @@ def fetch_trusted_domains_over_dbus(myapi, *keys, **options):
         fetch_domains_method = intf.get_dbus_method(
                 'fetch_domains',
                 dbus_interface=DBUS_IFACE_TRUST)
+        # Oddjobd D-BUS method definition only accepts fixed number
+        # of arguments on the command line. Thus, we need to pass
+        # remaining ones as ''. There are 30 slots to allow for extension.
+        method_arguments = [forest_name] + method_options
+        method_arguments.extend([''] * (30 - len(method_arguments)))
```

Comment:
Is it necessary to check that length is <= 30?

By the way, Python handles negative multiplication of a list just fine. I didn't know that before!

```
>>> [''] * -10
[]
```

Reply:
I don't think so. We could pass a total of four options, three of them take arguments, plus the trusted domain name. We construct the list ourselves, so this means 1+1+3*2=8 arguments total right now.